src/esrs/event_handler.rs

```diff
@@ -66,11 +80,20 @@ where
     Q: TransactionalEventHandler<A, E>,
     T: Deref<Target = Q> + Send + Sync,
 {
+    /// Deref call to [`TransactionalEventHandler::handle`].
     async fn handle(&self, event: &StoreEvent<A::Event>, executor: &mut E) -> Result<(), A::Error> {
         self.deref().handle(event, executor).await
     }
 }
 
+/// The `ReplayableEventHandler` trait is used to add the `replay` behavior on an `EventHandler`. Being
+/// replayable means that the operation performed by this `EventHandler` does not perform a side-effect
+/// (eg. an external API call).
+///
+/// The most common use case for this is when rebuilding read models: `EventHandler`s that write on
```

Comment:
Would be better to state this as: `Being replayable means that the operation performed by this `EventHandler` should be idempotent and should be intended to be "eventually consistent"`? Otherwise please help me stating this XD

Reply:
> Isn't writing to a DB a side-effect? :thinking:

~~It's pure except for IO~~

How about we combine the two? Something like

> Being replayable means that the operation performed by this `EventHandler` should be idempotent and should be
intended to be "eventually consistent"
In other words it means that they **should not** perform external API calls, generate random numbers or do anything that relies on external state and might change the outcome of this function.